### PR TITLE
Fix enum widths issue

### PR DESCRIPTION
**Pull request by rkapuscik**
Fixes cases where enum was limited to one bit size, even for larger declarations (reported [here](https://github.com/chipsalliance/f4pga-examples/issues/311)).

### CODE REVIEW

File: systemverilog-plugin/UhdmAst.cc [hunk 1]
```diff
@@ -1350,14 +1349,15 @@ void UhdmAst::move_type_to_new_typedef(AST::AstNode *current_node, AST::AstNode
...
-            }
             auto wire_node = new AST::AstNode(AST::AST_WIRE);
             wire_node->is_reg = true;
             wire_node->attributes["\\enum_type"] = AST::AstNode::mkconst_str(type_node->str);
             if (!type_node->children.empty() && type_node->children[0]->children.size() > 1) {
                 wire_node->children.push_back(type_node->children[0]->children[1]->clone());
+            } else {
+                // Add default range
+                auto range = make_range(31, 0);
+                wire_node->children.push_back(range);
```
Comment (kamilrakoczy) [inline systemverilog-plugin/UhdmAst.cc:1360]: Small nit, but I think it looks better.
```suggestion
                wire_node->children.push_back(make_range(31, 0));
```

**Review by kamilrakoczy — CHANGES_REQUESTED** [review]
2 small comments.

File: systemverilog-plugin/UhdmAst.cc (hunk 1 above)
Comment (rkapuscik) [reply systemverilog-plugin/UhdmAst.cc]: Changed.

File: systemverilog-plugin/UhdmAst.cc [hunk 2]
```diff
@@ -1895,6 +1873,10 @@ void UhdmAst::process_enum_const()
         constant_node->filename = current_node->filename;
         constant_node->location = current_node->location;
         current_node->children.push_back(constant_node);
+        auto left_const = AST::AstNode::mkconst_int(constant_node->range_left, true);
+        auto right_const = AST::AstNode::mkconst_int(constant_node->range_right, true);
+        auto range = make_ast_node(AST::AST_RANGE, {left_const, right_const});
+        current_node->children.push_back(range);
```
[the remark this answers is not in the document]
Comment (rkapuscik) [reply systemverilog-plugin/UhdmAst.cc]: Changed.

**Review by kamilrakoczy — APPROVED** [review]

**Pull request merged**